File: pkg/models/preferences.go
package models

type LocalePreference string

const (
	LocaleAuto               LocalePreference = "auto"
	LocaleEnglish            LocalePreference = "en"
	LocaleChineseSimplified  LocalePreference = "zh-CN"
	LocaleChineseTraditional LocalePreference = "zh-TW"
	LocaleSpanish            LocalePreference = "es"
	LocaleFrench             LocalePreference = "fr"
	LocaleGerman             LocalePreference = "de"
	LocalePortugueseBrazil   LocalePreference = "pt-BR"
	LocaleJapanese           LocalePreference = "ja"
	LocaleKorean             LocalePreference = "ko"
	LocaleHindi              LocalePreference = "hi"
	LocaleItalian            LocalePreference = "it"
)

// ThemePreference represents the UI theme preference.
type ThemePreference string

const (
	ThemePreferenceLight ThemePreference = "light"
	ThemePreferenceDark  ThemePreference = "dark"
	ThemePreferenceAuto  ThemePreference = "auto"
)

// TimezonePreference represents the preferred timezone display.
type TimezonePreference string

const (
	TimezonePreferenceLocal TimezonePreference = "local"
	TimezonePreferenceUTC   TimezonePreference = "utc"
)

// DisplayModePreference represents the preferred log display mode.
type DisplayModePreference string

const (
	DisplayModeTable   DisplayModePreference = "table"
	DisplayModeCompact DisplayModePreference = "compact"
	DisplayModeJSON    DisplayModePreference = "json"
)

// UserPreferences represents persisted user preferences.
type UserPreferences struct {
	Locale          LocalePreference      `json:"locale"`
	Theme           ThemePreference       `json:"theme"`
	Timezone        TimezonePreference    `json:"timezone"`
	DisplayMode     DisplayModePreference `json:"display_mode"`
	FieldsPanelOpen bool                  `json:"fields_panel_open"`
}

// UpdateUserPreferencesRequest represents a partial update to user preferences.
type UpdateUserPreferencesRequest struct {
	Locale          *LocalePreference      `json:"locale,omitempty"`
	Theme           *ThemePreference       `json:"theme,omitempty"`
	Timezone        *TimezonePreference    `json:"timezone,omitempty"`
	DisplayMode     *DisplayModePreference `json:"display_mode,omitempty"`
	FieldsPanelOpen *bool                  `json:"fields_panel_open,omitempty"`
}

File: frontend/src/api/preferences.ts
import { apiClient } from "./apiUtils";
import type { ThemeMode } from "@/stores/theme";

export type TimezonePreference = "local" | "utc";
export type DisplayModePreference = "table" | "compact";

export interface UserPreferences {
  theme: ThemeMode;
  timezone: TimezonePreference;
  display_mode: DisplayModePreference;
  fields_panel_open: boolean;
}

export interface UserPreferencesResponse {
  preferences: UserPreferences;
  is_default: boolean;
}

export type UpdateUserPreferencesRequest = Partial<UserPreferences>;

export const preferencesApi = {
  getPreferences: () => apiClient.get<UserPreferencesResponse>("/me/preferences"),
  updatePreferences: (data: UpdateUserPreferencesRequest) =>
    apiClient.put<UserPreferencesResponse>("/me/preferences", data),
};
